zetta_utils/layer/volumetric/index.py

```diff
@@ -19,44 +19,85 @@ class VolumetricIndex:  # pragma: no cover # pure delegation, no logic
     bcube: BoundingCube
     allow_slice_rounding: bool = False
 
+    @property
+    def start(self) -> IntVec3D:
+        return IntVec3D(*(e.start for e in self.to_slices()))
+
+    @property
+    def stop(self) -> IntVec3D:
+        return IntVec3D(*(e.stop for e in self.to_slices()))
+
+    @property
+    def shape(self) -> IntVec3D:
+        return self.stop - self.start
+
     def to_slices(self):
         return self.bcube.to_slices(self.resolution, self.allow_slice_rounding)
 
-    def pad(self, pad: IntVec3D):
+    def pad(self, pad: IntVec3D) -> VolumetricIndex:
```

Comment:
should we rename these guys to `with_pad`, `with_translation`? or maybe `padded`, `translated`?

Reply:
Yeah, I can do it.

Reply:
Actually, on second thought, I think this is fine as is. The threat if we switch to `with_pad` is that the user might think that the pad is an option that's set for whenever some operation is done (like a preprocessor).  Thoughts?

Reply:
Maybe `padded` then?
Are we going to switch to `with_index_processors` for immutable layers? If so, the `with_` thing can be consistent, and users will know what to expect.

Reply:
I think `padded` / `translated` is clearest for this usage (even outweighing the consistency). I'll change to those for both the bcube and the index.